pkg/spi/application/types.go

```diff
@@ -0,0 +1,33 @@
+package application
+
+import (
+	"github.com/docker/infrakit/pkg/types"
+)
+
+// Type is the type of an application.  This gives hint about what struct types to map to, etc.
+// It also marks one instance of an application as of different nature from another.
+type Type string
+
+//Operation : update operation code
+type Operation int
+
+const (
+	// TypeError is the type to use for sending errors in the transport of the events.
+	TypeError = Type("error")
+
+	// ADD new resources
+	ADD Operation = iota
+	// DELETE resources
+	DELETE
+	// UPDATE resources
+	UPDATE
+	// GET resources
+	GET
+)
```

Comment:
A few comments:

1. This SPI is pretty generic... are you intending for this to be a way to do generic CRUD operations on objects?    This makes me think about TPR in k8s: https://kubernetes.io/docs/concepts/ecosystem/thirdpartyresource/ 

2. The only problem here is that unless there's a backend store (like etcd), we can't really do much with this.   Let's say we now run in HA mode with etcd as backend, then this make sense.  

3. I think this should be a RESTful API listening on a port and *not* another JSON-RPC plugin interface.   This is so that this general API can be easily consumed by any application.  So you could deploy a web application that will automatically scale up and down the number of nodes and do rolling update when the application itself detects a new version is available.  Make sense?

4. What are objects that we are doing CRUD operations on?  Do you mean something like a group Spec or an instance Spec or something that maps to our plugin objects?   For example, if I wanted to let a developer write an application that can control its own node scaling, can we expose the Group spec as an object and let the user call this API to update the Group spec (e.g. changing the number of nodes)?